### PR TITLE
Add DateInterval type parameter

### PR DESCRIPTION
**Pull request by ossinkine**
This PR adds the ability to pass a DateInterval object as a query parameter

### CODE REVIEW

**Comment by Ocramius** [comment]
Looks good to me.

We need to plan something smarter for ORM v3 (maybe a replaceable infer mechanism?

**Comment by ossinkine** [comment]
Tests with the lowest dependencies didn't pass: https://travis-ci.org/doctrine/doctrine2/jobs/280750301
Should I increase the version in composer.json or add the version check in the class?

**Comment by Ocramius** [comment]
Bumping dependency version for 2.6.x is quite OK

**Review by lcobucci — APPROVED** [review]
LGTM 👍 

Thanks for the CS fixes as well 😄

File: tests/Doctrine/Tests/ORM/Query/ParameterTypeInfererTest.php [hunk 1]
```diff
@@ -13,17 +13,18 @@ class ParameterTypeInfererTest extends OrmTestCase
...
+            [new \DateInterval('P1D'), Type::DATEINTERVAL],
             [[2],          Connection::PARAM_INT_ARRAY],
             [["foo"],      Connection::PARAM_STR_ARRAY],
             [["1","2"],    Connection::PARAM_STR_ARRAY],
             [[],           Connection::PARAM_STR_ARRAY],
             [true,              Type::BOOLEAN],
-         ];
+        ];
 
         if (PHP_VERSION_ID >= 50500) {
```
Comment (lcobucci) [inline tests/Doctrine/Tests/ORM/Query/ParameterTypeInfererTest.php]: Unrelated, but we can simplify this since we require PHP 7.1+

**Comment by lcobucci** [comment]
@ossinkine 🚢

File: composer.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
         "php": "^7.1",
         "ext-pdo": "*",
         "doctrine/collections": "^1.4",
-        "doctrine/dbal": ">=2.5-dev,<2.7-dev",
+        "doctrine/dbal": "^2.6",
```
Comment (Majkl578) [inline composer.json]: Did we allow >=2.7-dev intentionally here or is it just accidental?

Comment (ossinkine) [reply composer.json]: I set it up intentionally for look at tests and they were passed, so I think it's ok.

Comment (lcobucci) [reply composer.json]: I think it was accidental =) It should indeed be `>=2.6-dev,<2.7-dev` or even `>=2.6-dev,<2.8-dev`

Comment (alcaeus) [reply composer.json]: With how carefully we're checking BC, does it even matter? Might as well use the semver operator here.

Comment (lcobucci) [reply composer.json]: That's true... @Majkl578 @Ocramius anything against that?

**Comment by Ocramius** [comment]
Nah, fine as-is

Marco Pivetta

http://twitter.com/Ocramius

http://ocramius.github.com/

**Pull request merged**